defs/go.bzl

```diff
@@ -16,34 +16,34 @@ load("@io_bazel_rules_go//go:def.bzl", "GoLibrary", "GoPath", "go_context", "go_
 
 def _compute_genrule_variables(resolved_srcs, resolved_outs, dep_import_paths):
     variables = {
-        "SRCS": cmd_helper.join_paths(" ", resolved_srcs),
-        "OUTS": cmd_helper.join_paths(" ", resolved_outs),
+        "SRCS": " ".join([src.path for src in resolved_srcs]),
+        "OUTS": " ".join([out.path for out in resolved_outs]),
         "GO_IMPORT_PATHS": " ".join(dep_import_paths),
     }
     if len(resolved_srcs) == 1:
-        variables["<"] = list(resolved_srcs)[0].path
+        variables["<"] = resolved_srcs[0].path
     if len(resolved_outs) == 1:
-        variables["@"] = list(resolved_outs)[0].path
+        variables["@"] = resolved_outs[0].path
     return variables
 
 def _go_genrule_impl(ctx):
```

Comment:
I've tested this on k/k; most notably, the `zz_generated.openapi.go` file remains unchanged.